### PR TITLE
Update Lighthouse Architecture Page

### PR DESCRIPTION
**Pull request by aswinsuryan**
The architecture is updated to the forward plugin based
design. The work-flow diagram is updated.

Fixes: submariner-io/submariner-website#152

Signed-off-by: Aswin Surayanarayanan <asuryana@redhat.com>

### CODE REVIEW

**Comment by mangelajo** [comment]
Aswin can you use the updated diagrams in : https://docs.google.com/presentation/d/1lu9Qjkka4whuTpSyrPUjrMQHiRq-9zI82xrdswyeBfg/edit#slide=id.g8593faed0f_2_17

Which don't mention etcd and instead the K8s API?

**Comment by aswinsuryan** [comment]
> Aswin can you use the updated diagrams in : https://docs.google.com/presentation/d/1lu9Qjkka4whuTpSyrPUjrMQHiRq-9zI82xrdswyeBfg/edit#slide=id.g8593faed0f_2_17
> 
> Which don't mention etcd and instead the K8s API?

Done.

File: src/content/architecture/service-discovery/_index.en.md [hunk 1]
```diff
@@ -4,38 +4,44 @@ date: 2020-03-02T21:25:35+01:00
...
 
 ![Lighthouse Architecture](/images/lighthouse/architecture.png)
 
 ### Lighthouse Agent
-The Lighthouse Agent runs in every cluster and accesses the Kubernetes API server running in the broker cluster to exchange service metadata information with other clusters. Local service information is exported to the broker and service information from other clusters is imported.
+The Lighthouse Agent runs in every cluster and accesses the Kubernetes API server running in
+the broker cluster to exchange service metadata information with other clusters. Local service
+information is exported to the broker and service information from other clusters is imported.
 
 #### WorkFlow
```
Comment (nyechiel) [inline src/content/architecture/service-discovery/_index.en.md:21]: nit: Workflow

File: src/content/architecture/service-discovery/_index.en.md [hunk 2]
```diff
@@ -4,38 +4,44 @@ date: 2020-03-02T21:25:35+01:00
...
-### Lighthouse Plugin
-Lighthouse plugin can be installed as an external plugin for CoreDNS, and will work along with the default Kubernetes plugin. It uses the MultiClusterService resources that are distributed by the controller for DNS resolution. The below diagram indicates a high-level architecture.
-
-![Lighthouse Plugin Architecture](/images/lighthouse/lighthousePluginArchitecture.png)
+### Lighthouse DNS Server
+The Lighthouse DNS server runs as an external DNS server which owns the domain supercluster.local.The
+KubeDNS is configured to forward any request supercluster.local to the Lighthouse DNS server, which
+uses the MultiClusterService resources that are distributed by the controller for DNS resolution
 
 #### WorkFlow
```
Comment (nyechiel) [inline src/content/architecture/service-discovery/_index.en.md:38]: Workflow

File: src/content/architecture/service-discovery/_index.en.md [hunk 3]
```diff
@@ -4,38 +4,44 @@ date: 2020-03-02T21:25:35+01:00
...
 The workflow is as follows.
 
-- A pod tries to resolve a Service name.
-- The Kubernetes plugin in CoreDNS will first try to resolve the request. If no records are present the request will be sent to Lighthouse plugin.
-- The Lighthouse plugin will use its MultiClusterService cache to try to resolve the request.
-- If a record exists it will be returned, otherwise the plugin will pass the request to the next plugin registered in CoreDNS.
+- A pod tries to resolve a Service name using the domain name supercluster.local
```
Comment (nyechiel) [inline src/content/architecture/service-discovery/_index.en.md:41]: s/pod/Pod

File: src/content/architecture/service-discovery/_index.en.md [hunk 4]
```diff
@@ -4,38 +4,44 @@ date: 2020-03-02T21:25:35+01:00
...
 <!-- Image Source: /images/lighthouse/source/controllerWorkFlow.vsdx  -->
 
-### Lighthouse Plugin
-Lighthouse plugin can be installed as an external plugin for CoreDNS, and will work along with the default Kubernetes plugin. It uses the MultiClusterService resources that are distributed by the controller for DNS resolution. The below diagram indicates a high-level architecture.
-
-![Lighthouse Plugin Architecture](/images/lighthouse/lighthousePluginArchitecture.png)
+### Lighthouse DNS Server
+The Lighthouse DNS server runs as an external DNS server which owns the domain supercluster.local.The
+KubeDNS is configured to forward any request supercluster.local to the Lighthouse DNS server, which
+uses the MultiClusterService resources that are distributed by the controller for DNS resolution
```
Comment (nyechiel) [inline src/content/architecture/service-discovery/_index.en.md:36]:
```suggestion
The Lighthouse DNS server runs as an external DNS server which owns the domain supercluster.local. KubeDNS is configured to forward any request sent to supercluster.local to the Lighthouse DNS server, which uses the MultiClusterService resources that are distributed by the controller for DNS resolution.
```

File: src/content/architecture/service-discovery/_index.en.md [hunk 5]
```diff
@@ -4,38 +4,44 @@ date: 2020-03-02T21:25:35+01:00
...
 
 #### WorkFlow
 The workflow is as follows.
 
-- A pod tries to resolve a Service name.
-- The Kubernetes plugin in CoreDNS will first try to resolve the request. If no records are present the request will be sent to Lighthouse plugin.
-- The Lighthouse plugin will use its MultiClusterService cache to try to resolve the request.
-- If a record exists it will be returned, otherwise the plugin will pass the request to the next plugin registered in CoreDNS.
+- A pod tries to resolve a Service name using the domain name supercluster.local 
+- The KubeDNS forwards the request to the Lighthouse DNS server.
```
Comment (nyechiel) [inline src/content/architecture/service-discovery/_index.en.md:42]:
```suggestion
- KubeDNS forwards the request to the Lighthouse DNS server.
```

**Comment by mangelajo** [comment]
@aswinsuryan and sorry being a little bit of a nitpick here, but is it possible to get the pngs with more resolution? the text is super blurry

File: src/content/architecture/service-discovery/_index.en.md (hunk 1 above)
Comment (aswinsuryan) [reply src/content/architecture/service-discovery/_index.en.md:21]: Done

File: src/content/architecture/service-discovery/_index.en.md (hunk 2 above)
Comment (aswinsuryan) [reply src/content/architecture/service-discovery/_index.en.md:38]: Done.

File: src/content/architecture/service-discovery/_index.en.md (hunk 4 above)
Comment (aswinsuryan) [reply src/content/architecture/service-discovery/_index.en.md:36]: Done.

File: src/content/architecture/service-discovery/_index.en.md [hunk 6]
```diff
@@ -4,38 +4,44 @@ date: 2020-03-02T21:25:35+01:00
 weight: 5
 ---
 
-The Lighthouse project provides DNS discovery for Kubernetes clusters connected by [Submariner](https://github.com/submariner-io/submariner) in multi-cluster environments. The solution is compatible with any CNI (Container Network Interfaces) plugin.
+The Lighthouse project provides DNS discovery for Kubernetes clusters connected by
+[Submariner](https://github.com/submariner-io/submariner) in multi-cluster environments.
+The solution is compatible with any CNI (Container Network Interfaces) plugin.
```
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply src/content/architecture/service-discovery/_index.en.md:9]: Done.

**Comment by aswinsuryan** [comment]
> @aswinsuryan and sorry being a little bit of a nitpick here, but is it possible to get the pngs with more resolution? the text is super blurry

Now it is full HD .. 8). Ya  that's right, I have updated it, it looks better for me , it  looks clear in Zoom as well. Please let me know if that is not the case.

File: src/content/architecture/service-discovery/_index.en.md (hunk 5 above)
Comment (aswinsuryan) [reply src/content/architecture/service-discovery/_index.en.md:42]: Done.

File: src/content/architecture/service-discovery/_index.en.md [hunk 7]
```diff
@@ -4,38 +4,43 @@ date: 2020-03-02T21:25:35+01:00
...
-- A pod tries to resolve a Service name.
-- The Kubernetes plugin in CoreDNS will first try to resolve the request. If no records are present the request will be sent to Lighthouse plugin.
-- The Lighthouse plugin will use its MultiClusterService cache to try to resolve the request.
-- If a record exists it will be returned, otherwise the plugin will pass the request to the next plugin registered in CoreDNS.
+- A Pod tries to resolve a Service name using the domain name supercluster.local 
+- KubeDNS forwards the request to the Lighthouse DNS server.
+- The Lighthouse DNS server will use its MultiClusterService cache to try to resolve the request.
+- If a record exists it will be returned, else an NXDomain error will be returned.
 
 ![Lighthouse CoreDNS WorkFlow](/images/lighthouse/coreDNSWorkFlow.png)
```
Comment (mangelajo) [inline src/content/architecture/service-discovery/_index.en.md:45]: Woops we keep referencing coreDNSWorkFlow.png here but the image has been deleted.

Comment (mangelajo) [reply src/content/architecture/service-discovery/_index.en.md:45]: ohh, sorry not deleted, just reduced in size as you said on slack

**Pull request merged**